frontend/src/pages/StreamPage.tsx

```diff
@@ -551,18 +556,23 @@ export function StreamPage() {
     return () => document.removeEventListener("keydown", handleKeyDown);
   }, [selectedTimelinePrompt]);
 
-  // Update temporal interpolation defaults when pipeline changes
+  // Update temporal interpolation defaults and clear prompts when pipeline changes
   useEffect(() => {
-    const pipeline = PIPELINES[settings.pipelineId];
+    const pipeline = pipelines?.[settings.pipelineId];
     if (pipeline) {
       const defaultMethod =
         pipeline.defaultTemporalInterpolationMethod || "slerp";
       const defaultSteps = pipeline.defaultTemporalInterpolationSteps ?? 4;
 
       setTemporalInterpolationMethod(defaultMethod);
       setTransitionSteps(defaultSteps);
+
+      // Clear prompts if pipeline doesn't support them
+      if (pipeline.supportsPrompts === false) {
```

Comment:
This is a bit of a hack. If the pipeline does not support prompts we just set the prompt to an empty string for now so that the timeline still works, but it doesn't show any text. This felt like the minimal change to get things working without requiring a larger refactor right now.

Reply:
I'm inclined to leave it for now to minimize drastic visual layout changes here and there are other things that having an indication of time progression may be useful for.